includes/abstracts/abstract.llms.shortcode.course.element.php

```diff
@@ -46,28 +46,40 @@ protected function get_default_attributes() {
 	 * $atts & $content are both filtered before being passed to get_output()
 	 * output is filtered so the return of get_output() doesn't need its own filter
 	 *
-	 * @return   string
-	 * @since    3.6.0
-	 * @version  3.6.0
+	 * @since 3.6.0
+	 * @version [version] Add fallback for editor block rendering.
+	 *
+	 * @return string
 	 */
 	protected function get_output() {
 
 		// Get a reference to the current page where the shortcode is displayed.
 		global $post;
-		$current_post = $post;
 
-		$course = get_post( $this->get_attribute( 'course_id' ) );
+		$current_id = $post->ID ?? null;
+
+		$id = $this->get_attribute( 'course_id' );
+
+		if ( ! $id ) {
+			$id = $current_id;
+		}
+
+		if ( ! $id && llms_is_editor_block_rendering() ) {
+			$id = LLMS_Shortcodes_Blocks::get_placeholder_course_id();
+		}
+
+		$course = get_post( $id );
 
 		// We don't have a post object to proceed with.
 		if ( ! $course ) {
 			return '';
 		}
 
-		if ( 'course' !== $course->post_type ) {
+		if ( in_array( $course->post_type, array( 'lesson', 'llms_quiz' ) ) ) {
```

Comment:
```suggestion
		if ( in_array( $course->post_type, array( 'lesson', 'llms_quiz' ), true ) ) {
```
what about assignments though? see the filter `llms_course_children_post_type` assignments add-on leverage as well.